### ISSUE TITLE
BO - Orders - impossible to extend order grid query from module

### ISSUE THREAD

**Issue by prestaalba** [PrestaShop/PrestaShop#24911]
#### Describe the bug

When you try to modify Orders list query from a module using actionOrderGridQueryBuilderModifier hook, always gets an error by using the "New client" filter.

#### Expected behavior

A module could extend orders list query using actionOrderGridQueryBuilderModifier hook with no errors.

#### Steps to Reproduce

1. Develop a module to add a new column on backoffice Orders grid using actionOrderGridQueryBuilderModifier hook.
2. Add a new data column to Orders grid (similar to example for Customers grid from official doc: https://devdocs.prestashop.com/1.7/modules/sample-modules/grid-and-identifiable-object-form-hooks-usage/#extending-grid-query-builder)
2. Click on backoffice Orders menu
3. Use "New client" filter with any value (Yes or No) and click on Search button
4. An error 500 is shown due to and incorrect table alias

**Screenshots**

- Module hooks code:
![1](https://user-images.githubusercontent.com/18466848/121567527-49093a00-c9e4-11eb-9d18-16fa920a64ad.JPG)
![2](https://user-images.githubusercontent.com/18466848/121567547-4dcdee00-c9e4-11eb-9236-9edd66e36a34.JPG)

hookActionOrderGridDefinitionModifier code:
```
public function hookActionOrderGridDefinitionModifier(array $params)
{
    $definition = $params['definition'];

    $definition->getColumns()->addAfter(
        'date_add',
        (new BooleanColumn('from_app'))
            ->setName($this->l('App order'))
            ->setOptions([
                'field' => 'from_app',
                'true_name' => $this->trans('Yes', [], 'Admin.Global'),
                'false_name' => $this->trans('No', [], 'Admin.Global'),
                'clickable' => true,
            ])
    );

    $definition->getFilters()->add(
        (new Filter('from_app', YesAndNoChoiceType::class))
        ->setTypeOptions([
            'required' => false,
        ])    
        ->setAssociatedColumn('from_app')
    );
}

public function hookActionOrderGridQueryBuilderModifier(array $params)
{
    $searchQueryBuilder = $params['search_query_builder'];

    $searchCriteria = $params['search_criteria'];

    $searchQueryBuilder->addSelect(
        'app.`id_order` IS NOT NULL AS `from_app`'
    );

    $searchQueryBuilder->leftJoin(
        'o',
        '`app_order_table`',
        'app',
        'app.`id_order` = o.`id_order`'
    );

    if ('from_app' === $searchCriteria->getOrderBy()) {
        $searchQueryBuilder->orderBy('app.`id_order`', $searchCriteria->getOrderWay());
    }

    foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
        if ('from_app' === $filterName) {
            $searchQueryBuilder->andWhere('app.`id_order` IS ' . ($filterValue ? 'NOT' : '') . ' NULL');
        }
    }
}
```
- Error details:
![3](https://user-images.githubusercontent.com/18466848/121567574-58888300-c9e4-11eb-848c-f746a80b4062.JPG)

#### Additional information

A further investigation show that error came from leftJoin alias used on $searchQueryBuilder modification. The problem is that standard "o" alias completly change on query builder when a "New client" filter is used. This is caused by an incorrect subquery that do not use standard "o" alias (in fact do not use any alias at all) on https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Grid/Query/OrderQueryBuilder.php#L275. This makes impossible to extend orders query using a join stament unless you remove "New client" filter and column from grid.

* PrestaShop version: develop
* PHP version: N/A

**Comment by matks** [comment]
That is a great and detailed bug report! Thank you

**Comment by florine2623** [comment]
Hello @prestaalba ,

Can you provide an example module that does that use actionOrderGridQueryBuilderModifier hook ?
That will help us in the process of testing your issue.

Thanks!

**Comment by prestaalba** [comment]
Hello @florine2623 , I cannot provide a module but I already provided all the necessary code and examples for testing. Regards

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/prestashop-core-developers could someone please try to reproduce the issue :pray: 
Sorry for the inconvenience and thanks in advance :relaxed:

**Comment by mrbig00** [comment]
This is caused by [this](https://github.com/PrestaShop/PrestaShop/blob/ad181f7fb89dea67dfebba779a6440854baa6d5b/src/Core/Grid/Query/OrderQueryBuilder.php#L266), honestly I don't even know why it is creating a temp table instead using a subquery or a having condition?!

**Comment by atomiix** [comment]
@hibatallahAouadni I confirm I have been able to reproduce this issue!

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team this issue is a regression since `1.7.7.0`. So, what we should do? should we add the label `regression` and put this issue in `Project PrestaShop`? or it's just a bug and put it in the EPIC `Hooks`?

**Comment by MatShir** [comment]
All the grid should be extendable, the issue should be shipped into the next patch 🚀

**Comment by Progi1984** [comment]
Fixed by @atomiix in #26221
